amy/workshops/models.py

```diff
@@ -560,6 +634,31 @@ def inhouse_instructor_training_seats_remaining(self) -> int:
         c = self.inhouse_instructor_training_seats_rolled_over or 0
         return a - b - c
 
+    @property
+    def cldt_seats_total(self) -> int:
+        """Calculate combined CLDT seats total.
+
+        Unlike workshops w/o admin fee, CLDT seats have two numbers
+        combined to calculate total of allowed seats in CLDT events.
+        """
+        a = self.cldt_seats
+        b = self.additional_cldt_seats
+        c = self.cldt_seats_rolled_from_previous or 0
+        return a + b + c
+
+    @cached_property
+    def cldt_seats_utilized(self) -> int:
+        """Count number of learner tasks that point to this membership."""
+        return self.task_set.filter(role__name="learner", seat_public=False).count()
```

Comment:
Since this property should mimick the query in `MembershipManager.annotate_with_seat_usage`, there's additional filtering missing:

1. `cld_seats_utilized()` should also filter by event tag CLDT
2. in `MembershipManager.annotate_with_seat_usage`, `_cldt_seats_utilized` should also filter by `seat_public`
3. in `MembershipManager.annotate_with_seat_usage`, `_cldt_seats_remaining` should filter by `seat_public` in the same way (currently it uses `public=True`)
4. both `public_instructor_training_seats_utilized()` and `inhouse_instructor_training_seats_utilized()` should also filter by TTT tags (TTT, ITT)

Generally speaking the model properties should come down to the same equations as in annotations.

Reply:
Updated all calculations and set CLDT to public=False as these events are member based and not open to the public.